Doctrine/DoctrineJobTimingManager.php

```diff
@@ -45,6 +45,6 @@ abstract protected function persist($object, $action = 'persist');
 
     public function pruneJobTimings(\DateTime $olderThan)
     {
-        return $this->removeOlderThan($this->getJobTimingClass(), 'createdAt', $olderThan);
+        return $this->removeOlderThan($this->getJobTimingClass(), 'finishedAt', $olderThan);
```

Comment:
This slightly changes behavior here. What happens if there are very old jobs that never finish.

Frankly it's been a while since I traced out the code here, but that's one concern I have with changing this from createdAt to finishedAt.

Reply:
> What happens if there are very old jobs that never finish.

After looking a bit deeper, I don't think this is an issue. It seems that Jobs that never finish won't create JobTiming records. 

JobTiming objects don't seem to be relationally associated anywhere, and as such, there is no means to modify a JobTiming. The only instantiation I can find for one seems to be in the PerformRecording function within the TimingManger. A new JobTiming is created every time PerformRecording is called, and there is no return of the instantiation. Essentially, any record in the JobTiming table should be purgeable without risk to data integrity. 

I've included the function below for quick reference:

```
public function performRecording($status, \DateTime $finishedAt = null)
    {
        if (null === $finishedAt) {
            $finishedAt = \Dtc\QueueBundle\Util\Util::getMicrotimeDateTime();
        }

        /** @var JobTiming $jobTiming */
        $jobTiming = new $this->jobTimingClass();
        $jobTiming->setFinishedAt($finishedAt);
        $jobTiming->setStatus($status);
        $objectManager = $this->getObjectManager();
        $objectManager->persist($jobTiming);
        $objectManager->flush();
    }
```